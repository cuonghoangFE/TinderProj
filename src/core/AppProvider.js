/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

// AppProvider.js
import React, {useContext} from 'react';
import Storage from '../utils/storage';
import userAction from '../actions/user';

const AppContext = React.createContext();

export function withApp(Component) {
  return props => {
    return <Component {...props} {...useContext(AppContext)} />;
  };
}

const AppContextProvider = ({children}) => (
  <AppContext.Provider
    value={{
      userStore: userAction.userStore({}),
      myFavoritesStore: Storage.useAsyncStorage('myFavorites', []),
    }}>
    {children}
  </AppContext.Provider>
);

export default AppContextProvider;
