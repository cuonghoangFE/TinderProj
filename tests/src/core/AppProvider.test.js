import React from 'react';
import renderer from 'react-test-renderer';
import {withApp} from '../../../src/core/AppProvider.js';

describe('AppProvider', () => {
  describe('withApp HOC', () => {
    it('renders correctly', () => {
      const Com = withApp(React.Fragment);
      const tree = renderer.create(<Com />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
