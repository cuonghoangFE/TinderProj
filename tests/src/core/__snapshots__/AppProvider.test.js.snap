// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppProvider withApp HOC renders correctly 1`] = `null`;
